syntax = "proto3";

service Matchmaker {

  rpc newGame (NewGameRequest) returns (NewGameReply) {}

  rpc listGame (ListGameRequest) returns (stream ListGameReply) {}

  rpc findGame (FindGameRequest) returns (stream FindGameReply) {}

  rpc deleteGame (DeleteGameRequest) returns (DeleteGameReply) {}

  rpc saveGame (SaveGameRequest) returns (SaveGameReply) {}
}

message NewGameRequest {
  string gameId = 1;
  string playerId = 2;
}

message NewGameReply {
  bool success = 1;
}

message ListGameRequest {}

message ListGameReply {
  string address = 1;
  string gameId = 2;
  string playerId = 3;
}

message FindGameRequest {
  string gameId = 1;
  string playerId = 2;
}

message FindGameReply {
  string address = 1;
  string gameId = 2;
  string playerId = 3;
}

message DeleteGameRequest {
  string gameId = 1;
  string playerId = 2;
}

message DeleteGameReply {
  bool success = 1;
}

message SaveGameRequest {}

message SaveGameReply {
  bool success = 1;
}
